### ISSUE TITLE
insert_n cannot (currently) ever be constexpr

### ISSUE THREAD

**Issue by gnzlbg** [everard/contiguous-container#8]
The implementation of `contiguous_container::insert_n` is marked as `constexpr` but it uses standard library functions like `std::min`,`std::advance`, `std::move_backward`, ... that are not `constexpr` in C++14 and, at least for `advance` and `move_backward`, are not constexpr in C++17 either. 

Making it `constexpr` should be however doable by reimplementing the `std` functionality in a constexpr friendly manner.

**Comment by everard** [comment]
Yes. When these functions become constexpr in later versions of Standard (hopefully), then all insert functionality will also become usable in constexpr context.
I don't see a reason to implement all needed algorithms, though.

**Comment by gnzlbg** [comment]
> I don't see a reason to implement all needed algorithms, though.

The implementation fails in a constexpr context. If the objective of your implementation is to prove that the proposal is implementable then... that is all the reason you should need: the current implementation does not prove it.

Whether the implementation uses `std::min/advance/move_backward` or something else is an implementation detail, so there is actually nothing preventing the implementation from becoming `constexpr`.

> When these functions become constexpr in later versions of Standard **(hopefully)**, then all insert functionality will also become usable in constexpr context.

This argument raises questions like "If this cannot be `constexpr` till later, and it can be safely marked as `constexpr` later, why don't we mark it as constexpr later when we know that it will work?".

I think that you took the right decision by marking it constexpr, i just also think that the implementation should show that doing so works just fine.

**Comment by everard** [comment]
Ok, got it
